frontend/src/scenes/billing/BillingProductPricingTable.tsx

```diff
@@ -123,6 +136,19 @@ export const BillingProductPricingTable = ({
                                           0
                                       ) || 0
                                     : 0)
+
+                      const projectedTotalWithLimitForTier =
+                          product.type === 'session_replay'
+                              ? parseFloat(tier.projected_amount_usd_with_limit || '0')
+                              : (parseFloat(tier.projected_amount_usd_with_limit || '') || 0) +
+                                ('addons' in product
+                                    ? product.addons?.reduce(
+                                          (acc: number, addon: BillingProductV2AddonType) =>
+                                              acc + (parseFloat(addon.tiers?.[i]?.projected_amount_usd || '') || 0),
+                                          0
+                                      ) || 0
+                                    : 0)
```

Comment:
Also I know this is an existing pattern but would be awesome to have a util with tests for this as we re-use it at least ~3 times here and it makes it a bit hard to read. Even better if these were just selectors in `billingProductLogic`